security.md

```diff
@@ -0,0 +1,97 @@
+# Security and privacy considerations
+
+## Code interception attacks
+
+By default, Wormhole codes contain 16 bits of entropy. Failed attempts of
+guessing a code will cause both clients to error out. Thus, an attacker
+has a one-in-65536 chance of successfully guessing the code, while being
+detected in all other cases.
+
+If the attacker successfully guesses the code, they can attempt a
+machine-in-the-middle (MitM) attack. For this, they quickly reconnect with
+the same code in order to connect with the second peer. With that, they get a
+secure channel to each side while being able to read and modify the connection's
+content.
+
+Both cases can be mitigated by enabling the verifier feature, in which case
+even successful attacks can be detected.
+
+In general, this is a very similar process to ordinary key exchanges like
+Diffie-Hellman (or variations), except that MitM attackers additionally have to
+guess the code correctly. (Hence the name, "*password authenticated* key
+exchange", PAKE.)
+
+### Password entropy
+
+Compared to password strengths we are used to nowadays, 16 bits may seem scarily
+small, even knowing that an attacker only has one try to guess the password. But
+to put it into perspective: As a user, you'd have to experience 32000 failed
+transfers because of an attack in order to having a 50% chance of a file being
+intercepted. From an attacker's perspective, guessing thousands of codes is not
+difficult, but there is no possibility of targetting any individual connection,
+making any gained data fairly useless. From a service perspective, any successful
+attack implies 65536 failed attempts on average, which will quickly be noticed
```

Comment:
I think you mean ~32000 here (to match above)...?

Reply:
Nope, should be correct. The one is the total average, the other one is only up to 50% probability. Something something Birthday paradox
(The probability of having a cracked password after 65536 attempts is less than one, but the probability of needing less than that also exists so it exactly cancels out on average)

Reply:
Can you explain further? I don't immediately see how the Birthday Paradox is relevant here?

Reply:
Okay thought about this more and it's not the Birthday paradox. So IMO 1/65536 still is correct, but one needs a different calculation for the other number.

I end up at (1 − 1/65536)^45000 = 0.503258791 so if the calculation is correct one would need 45000 guesses for a 50% success probability.

Reply:
After 32768 attempts, you have an *expected* number of successful guesses that is ``0.5``. However, this number also includes events where you guess the right key multiple times, which doesn't really help you as an attacker. Therefore, the number of attempts you need to have a 50% chance of getting *at least* one correct guess is higher (and seems to be around ``45000``, as @piegamesde showed).

Reply:
Addendum: this still works even in our model where the attacker only gets a guess at each code. In that case, I think it becomes "the probability of (not) being hacked after 45000 failed exchanges is 0.5"

So in the end I'd say both numbers are correct, they just answer different questions.

Reply:
@Fxrh that part I don't understand: there's only one guess available per setup attempt, so what does "guess the right key multiple times" mean here..?

Reply:
I think the text could still benefit from more words, then? (And 65536 isn't either of these numbers, so that should change, yes?)